modules/porous_flow/src/actions/PorousFlowSinglePhaseBase.C

```diff
@@ -57,14 +65,27 @@ PorousFlowSinglePhaseBase::PorousFlowSinglePhaseBase(const InputParameters & par
     _fp(getParam<UserObjectName>("fp")),
     _biot_coefficient(getParam<Real>("biot_coefficient")),
     _add_darcy_aux(getParam<bool>("add_darcy_aux")),
-    _add_stress_aux(getParam<bool>("add_stress_aux"))
+    _add_stress_aux(getParam<bool>("add_stress_aux")),
+    _use_brine(getParam<bool>("use_brine")),
+    _nacl_index(getParam<unsigned>("nacl_index"))
 {
   if ((_coupling_type == CouplingTypeEnum::ThermoHydro ||
        _coupling_type == CouplingTypeEnum::ThermoHydroMechanical) &&
       _temperature_var.size() != 1)
     mooseError("PorousFlowSinglePhaseBase: You need to specify a temperature variable to perform "
                "non-isothermal simulations");
 
+  if (_use_brine && !params.isParamValid("mass_fraction_vars"))
+    mooseError("PorousFlowSinglePhaseBase: You need to specify at least one component in "
```

Comment:
It would be good to add 3 exception tests to check your 3 mooseErrors

Reply:
Done